components/base/base.go

```diff
@@ -49,13 +49,15 @@ type Base interface {
 
 	// Spin spins the robot by a given angle in degrees at a given speed.
 	// If a speed of 0 the base will stop.
+	// A positive angle is measured in degrees to the left.
+	// A positive speed turns to the left.
```

Comment:
eeeeeee, so...

Spin and movestraight are a little tricky.

+ve angle +ve speed -> turn to left
-ve angle -ve speed -> turn to left
-ve angle +ve speed -> turn to right
+ve angle -ve speed -> turn to right

Reply:
And for MoveStriaght:

+ve angle +ve speed -> forwards
-ve angle -ve speed -> forwards
-ve angle +ve speed -> backwards
+ve angle -ve speed -> backward

Reply:
Whaaaat why does negative angle positive speed turn right? It seems like it should be the opposite

And move straight doesn't take an angle--I assume you mean distance in place of angle?

Reply:
It's the result of having no directionality in our motor API, which is what was chosen a year or so ago.

Reply:
so if I wan to turn clockwise (negative angle) 15degs at a speed of 3degs per second, I expect to turn clockwise.

if I want to turn clockwise (negative angle), at a speed of -15degs per sec, that cancels out. This behaviour with speed is something I believe we can change if it's too confusing. It's not like the input for SetVelocity.

Reply:
if you'd like something for now - how about: "given a positive speed and a positive angle, the base  turns to the left"

Reply:
and for Movestriaght: "given a positive distance and a positive speed, a base moves forwards"

Reply:
So, what if I want my base to turn clockwise 405 degrees at 10 deg/s? -405, 10? And it'd ultimately land at 45 degrees to the right of where it was originally facing? But if I gave it 405, -10, it would do the same thing? Or would it land somewhere different, like 45 degrees to the left of where it was originally facing? It seems like if those are both the same, the latter is unintuitive, and maybe the degrees should have to  be >0 so people don't get confused.

Reply:
```suggestion
	// Given a positive speed and a positive angle, the base turns to the left.
```